internal/volumes/volumes.go

```diff
@@ -329,8 +421,15 @@ func GetBindMount(sys *types.SystemContext, args []string, contextDir string, st
 
 	overlayDir := ""
 	if mountedImage != "" || mountIsReadWrite(newMount) {
-		if newMount, overlayDir, err = convertToOverlay(newMount, store, mountLabel, tmpDir, 0, 0); err != nil {
-			return newMount, "", "", "", err
+		switch store.GraphDriverName() {
+		case "overlay":
```

Comment:
(There is precedent for hard-coding a condition like this … but maybe the checks could all be consolidated into a single function.)